src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Logging/ProjectLoggerExtensions.ProjectLoggerBatch.cs

```diff
@@ -45,43 +43,22 @@ public void WriteLine(in StringFormat format)
             {
                 if (IsEnabled)
                 {
-                    Init();
-                    WriteNewLineIfNeeded();
-                    WriteIndent();
-                    Write(format);
```

Comment:
I inlined these methods as it allows the compiler to flow information about the null state of `_builder` through all methods.